### PR TITLE
Pdct 1221/remove vestige of organisation

### PR DESCRIPTION
**Pull request by annaCPR**
# What's changed
- removed organisation radio buttons from the Collection form (both create and edit) as the value should be taken from the user context
- do not send the organisation field when creating a collection as it is not used in the backend
- added a test to cover creating of a new collection

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

### CODE REVIEW

File: src/components/forms/CollectionForm.tsx [hunk 1]
```diff
@@ -58,11 +47,13 @@ export const CollectionForm = ({ collection: loadedCollection }: TProps) => {
     const collectionData: ICollectionFormPost = {
       title: collection.title,
       description: collection.description,
-      organisation: collection.organisation as TOrganisation,
```
Comment (katybaulch) [inline src/components/forms/CollectionForm.tsx:61]: `organisation` is a required field in the admin backend, so whatever is in the final object to be posted must contain an `organisation` field. You can decode the user token (like we do in the family form) to get the organisation to pass back to the backend until we have a change to remove the vestiage of organisation from the admin-backend DTOs though

**Review by katybaulch — CHANGES_REQUESTED** [review]
love that you've pulled the handlers out for mocking! looking good, but I think we need to keep `organisation` as a not nullable/undefined field as that's what the admin backend expects. Saying that, if @PatFawbertMills is happy this is all good with the model in the backend, I'm happy

File: src/interfaces/Collection.ts [hunk 2]
```diff
@@ -11,5 +11,5 @@ export interface ICollection {
 export interface ICollectionFormPost {
   title: string
   description?: string
-  organisation: TOrganisation
+  organisation?: TOrganisation
```
[the remark this answers is not in the document]
Comment (katybaulch) [reply src/interfaces/Collection.ts:14]: Whoops - I can see it isn't required for create after all. My bad

File: src/components/forms/CollectionForm.tsx [hunk 3]
```diff
@@ -58,11 +47,13 @@ export const CollectionForm = ({ collection: loadedCollection }: TProps) => {
...
     }
 
     if (loadedCollection) {
-      return await updateCollection(collectionData, loadedCollection.import_id)
+      return await updateCollection(
```
Comment (katybaulch) [inline src/components/forms/CollectionForm.tsx:53]: Ah I can see we're loading the org here - all good

File: src/components/forms/CollectionForm.tsx (hunk 1 above)
Comment (katybaulch) [reply src/components/forms/CollectionForm.tsx:61]: Have just seen this snippet - so all good my end. Will approve!
https://github.com/climatepolicyradar/navigator-admin-frontend/pull/99/files#r1705679747

**Review by katybaulch — APPROVED** [review]
lgtm - thanks for your patience!

File: src/components/forms/CollectionForm.tsx (hunk 1 above)
Comment (annaCPR) [reply src/components/forms/CollectionForm.tsx:61]: This only moves the assignment of value to organisation to lower in the code where we decide whether it's an update or create so we can reuse the same object. You can see a bit lower down that I add organisation from the loadedFamily to this object when it's an update. 
A ticket to remove organisation from the backend DTO can be found [here](https://linear.app/climate-policy-radar/issue/PDCT-1304/remove-organisation-field-from-collectiondto)

**Pull request merged**